pandas/core/dtypes/cast.py

```diff
@@ -255,9 +255,12 @@ def changeit():
 
 
 def maybe_promote(dtype, fill_value=np.nan):
-
     # if we passed an array here, determine the fill value by dtype
-    if isinstance(fill_value, np.ndarray):
+    if is_extension_array_dtype(dtype):
+        # XXX: verify this change
+        fill_value = dtype.na_value
```

Comment:
this should only be done when `fill_value` is NaN? (similar check for datetime-likes to overwrite it with iNaT) 

As when you fill with other values, it should still check it?